Add generated opcheck test for if the pt2_compliant_tag is incorrectly applied

zou3519 opened the pull request:
Summary:
If there are xfails in the failures_dict and the operator has the
pt2_compliant_tag, then we raise an error. These generated tests are separate
from those in the failures dict because we don't actually need any sample
inputs to check this.

Test Plan: - New tests

Differential Revision: D50936201

zou3519 commented:
NB: test failures all look unrelated (I'm not really sure how to do an export from internal to a reasonable commit base)

ezyang commented on test/test_custom_ops.py:
```diff
@@ -1819,10 +1819,16 @@ def op_with_incorrect_schema(testcase, name):
+        try:
+            return torch._library.utils.lookup_op(qualname)
+        except AttributeError:
+            pass
```
hmm?

zou3519 replied:
Fixed, I was being lazy. We needed a way to initialize the operator once

ezyang commented on torch/testing/_internal/optests/generate_tests.py:
```diff
@@ -253,6 +254,54 @@ def new_method(*args, **kwargs):
+        definitely_not_pt2_compliant = len(xfailed_tests) > 0
+        generated = generate_test(qualname, definitely_not_pt2_compliant, xfailed_tests)
+
+        # Could result in collisions, but unlikely
```
so are you going to check that you are colliding? Seems relatively cheap to do when you setattr

zou3519 replied:
Yeah we can do that

ezyang reviewed (approved):
okey dokey

zou3519 committed:
Add generated opcheck test for if the pt2_compliant_tag is incorrectly applied (#112759)

Summary:

If there are xfails in the failures_dict and the operator has the
pt2_compliant_tag, then we raise an error. These generated tests are separate
from those in the failures dict because we don't actually need any sample
inputs to check this.

Test Plan: - New tests

Reviewed By: ezyang

Differential Revision: D50936201
